In [1]:
from transformers import AutoTokenizer, AutoModelForTokenClassification
from transformers import pipeline

2024-05-08 00:44:01.706402: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-05-08 00:44:01.706544: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-05-08 00:44:01.888148: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered


In [2]:
tokenizer = AutoTokenizer.from_pretrained("dslim/bert-base-NER")
model = AutoModelForTokenClassification.from_pretrained("dslim/bert-base-NER")

tokenizer_config.json:   0%|          | 0.00/59.0 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/829 [00:00<?, ?B/s]

vocab.txt:   0%|          | 0.00/213k [00:00<?, ?B/s]

added_tokens.json:   0%|          | 0.00/2.00 [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/112 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/433M [00:00<?, ?B/s]

Some weights of the model checkpoint at dslim/bert-base-NER were not used when initializing BertForTokenClassification: ['bert.pooler.dense.bias', 'bert.pooler.dense.weight']
- This IS expected if you are initializing BertForTokenClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForTokenClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).


In [3]:
nlp = pipeline("ner", model=model, tokenizer=tokenizer)
example = "My name is Wolfgang and I live in Berlin"

In [4]:
ner_results = nlp(example)
print(ner_results)

[{'entity': 'B-PER', 'score': 0.9990139, 'index': 4, 'word': 'Wolfgang', 'start': 11, 'end': 19}, {'entity': 'B-LOC', 'score': 0.999645, 'index': 9, 'word': 'Berlin', 'start': 34, 'end': 40}]


[
    {
        'entity': 'LABEL_0', 
        'score': 0.9992238, 
        'index': 1, 
        'word': 'My', 
        'start': 0, 
        'end': 2
    }, 

    {
        'entity': 'LABEL_0', 
        'score': 0.9994271, 
        'index': 2, 
        'word': 'name', 
        'start': 3, 
        'end': 7
    }, 

    {
        'entity': 'LABEL_0', 
        'score': 0.99953353, 
        'index': 3, 
        'word': 'is', 
        'start': 8, 
        'end': 10
    }, 
    
    {
        'entity': 'LABEL_1', 
        'score': 0.99110633, 
        'index': 4, 
        'word': 'Wolfgang', 
        'start': 11, 
        'end': 19
    }, 
    
    {
        'entity': 'LABEL_0', 
        'score': 0.9994848, 
        'index': 5, 
        'word': 'and', 
        'start': 20, 
        'end': 23
    }, 
    
    {
        'entity': 'LABEL_0', 
        'score': 0.9873789, 
        'index': 6, 
        'word': 'I', 
        'start': 24, 
        'end': 25
    }, 
    
    {
        'entity': 'LABEL_0', 
        'score': 0.9990946, 
        'index': 7, 
        'word': 'live', 
        'start': 26, 
        'end': 30
    }, 
    
    {
        'entity': 'LABEL_0', 
        'score': 0.9990859, 
        'index': 8, 
        'word': 'in', 
        'start': 31, 
        'end': 33
    }, 
    
    {
        'entity': 'LABEL_5', 
        'score': 0.9967968, 
        'index': 9, 
        'word': 'Berlin', 
        'start': 34, 
        'end': 40
    }
]

[
    {
        'entity': 'B-PER', 
        'score': 0.9990139, 
        'index': 4, 
        'word': 'Wolfgang', 
        'start': 11, 
        'end': 19
    }, 
    
    {
        'entity': 'B-LOC', 
        'score': 0.999645, 
        'index': 9, 
        'word': 'Berlin', 
        'start': 34, 
        'end': 40
    }
]